File: src/ClimbingMap.Domain/Entities/Route.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ClimbingMap.Domain.Entities {
   public class Route {
      public string Id { get; set; }
      public Location Location { get; set; } = new Location();
      public double Difficulty { get; set; }
      public Grade Grade { get { return new Grade(Difficulty); } }
      public string Name { get; set; }
      public string Info { get; set; }
      public string[] Tags { get; set; } = new string[0];
      public string[] Links { get; set; } = new string[0];
      public Schema[] Schemas { get; set; } = new Schema[0];
   }
}

File: src/web/boulder-guide/src/app/domain/Route.ts
import { Topo } from './Topo'
import { Video } from './Video'
import { Location } from './Location'

export class Route {
   Id!: string;
   Location!: Location;
   Difficulty!: number;
   Name!: string;
   Info!: string;
   Tags!: string[];
   Links!: string[];
   Videos!: Video[];
   Schemas!: Topo[];
}
